src/hooks/useFirestore.js

```diff
@@ -3,21 +3,24 @@ import { db } from '../lib/firebase';
 
 const useFirestore = (collection) => {
   const [docs, setDocs] = useState([]);
+  const [loading, setLoading] = useState(false);
 
   useEffect(() => {
     const unsubscribe = db.collection(collection).onSnapshot((snapshot) => {
       let documents = [];
       snapshot.forEach((doc) => {
         documents.push({ ...doc.data(), id: doc.id });
       });
+      setLoading(true);
```

Comment:
I think we may want to throw this initial `setLoading` to be the first line inside the `useEffect` callback. (i.e. 
``useEffect(() => {
   setLoading(true);
  const unsubscribe = ...
}, [collection])